pkg/client/transport/transport.go

```diff
@@ -55,9 +55,6 @@ func TLSConfigFor(c *Config) (*tls.Config, error) {
 	if !(c.HasCA() || c.HasCertAuth() || c.TLS.Insecure) {
 		return nil, nil
 	}
-	if c.HasCA() && c.TLS.Insecure {
```

Comment:
off the top of my head, in DirectClusterConfig.getCluster(), after merging all configs together:

```
	// an override of --insecure-skip-tls-verify=true and no accompanying CA/CA data should clear already-set CA/CA data
	// otherwise, a kubeconfig containing a CA reference would return an error that "CA and insecure-skip-tls-verify couldn't both be set"
	if config.overrides.ClusterInfo.InsecureSkipTLSVerify && len(config.overrides.ClusterInfo.CertificateAuthority) == 0 && len(config.overrides.ClusterInfo.CertificateAuthorityData) == 0 {
		mergedClusterInfo.CertificateAuthority = ""
		mergedClusterInfo.CertificateAuthorityData = nil
	}
```